### PR TITLE
[TBCM-2] Create Side Menu Bar Component

### PR DESCRIPTION
**Pull request by leon-fw**
- Tried to make it visually as close to the Figma design as possible
- Collapses/Expands as per Figma design
- Active buttons are highlighted

To-do:
- Refactoring
- The buttons should ultimately be links to whatever page, need  to integrate that later
- Maybe some further spacing/styling on the components

### CODE REVIEW

File: apps/web/src/components/SidebarData.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { SidebarButtonProps, ButtonKind } from './Interfaces';
+import { faClipboardList, faUsers } from '@fortawesome/free-solid-svg-icons';
+
+export const SidebarData: SidebarButtonProps[] = [
```
Comment (karthikktamilmani-freshworks) [inline apps/web/src/components/SidebarData.tsx:4]: Great work. Could you rename this to SidebarNavData

File: apps/web/src/components/SidebarCollapsible.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      >
+        {faIcon ? (
+          <FontAwesomeIcon
+            className={`${active ? 'text-gray-100' : 'text-gray-400'} w-6 h-6`}
+            icon={faIcon}
+          />
+        ) : null}
+
+        <span className={`${open ? 'hidden' : ''} ${active ? 'text-gray-100' : 'text-gray-400'}`}>
+          Planning
```
Comment (karthikktamilmani-freshworks) [inline apps/web/src/components/SidebarCollapsible.tsx:35]: Think we're getting the text from the parent. We wouldn't need this hardcoding

File: apps/web/tailwind.config.js [hunk 3]
```diff
@@ -37,6 +37,7 @@ module.exports = {
         bcDisabled: '#CFCFCF',
         bcYellowCream: '#F9F1C6',
         bcYellowCreamStroke: '#FAEBCC',
+        bcDarkBlue: '#001a33',
```
Comment (karthikktamilmani-freshworks) [inline apps/web/tailwind.config.js:40]: Can we make sure the color matches the standard here?

File: apps/web/src/components/SidebarData.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { SidebarButtonProps, ButtonKind } from './Interfaces';
```
Comment (karthikktamilmani-freshworks) [inline apps/web/src/components/SidebarData.tsx:1]: Could we remove this file and move it to src/common/constants.ts

File: apps/web/src/components/Interfaces.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { IconDefinition } from '@fortawesome/fontawesome-common-types';
+import { Dispatch, SetStateAction } from 'react';
+
+export enum ButtonKind {
```
[the remark this answers is not in the document]
Comment (karthikktamilmani-freshworks) [reply apps/web/src/components/Interfaces.tsx:4]: Maybe SidebarButtonKind

[the remark this answers is not in the document]
Comment (leon-fw) [reply apps/web/src/components/Interfaces.tsx:4]: Excellent, thank you, will rename it into that!

File: apps/web/tailwind.config.js (hunk 3 above)
Comment (leon-fw) [reply apps/web/tailwind.config.js:40]: Khan confirmed that this meets the BC guidelines, leaving it as is.

File: apps/web/src/components/Interfaces.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (leon-fw) [reply apps/web/src/components/Interfaces.tsx:4]: Renamed

File: apps/web/src/components/SidebarData.tsx (hunk 4 above)
Comment (leon-fw) [reply apps/web/src/components/SidebarData.tsx:1]: Moved to src/common/constants.ts, thanks Karthikk!

**Review by karthikktamilmani-freshworks — APPROVED** [review]

**Review by PeterPavlatos — APPROVED** [review]

**Pull request merged**